### PR TITLE
Integrate renku-gateway

### LINKED ISSUE #313: Integrate renku-gateway
Integrate the gateway component from https://github.com/SwissDataScienceCenter/renku-gateway.

The gateway adds the following features:
- Login routes for clients
- Forwarding requests to GitLab/Storage while exchanging tokens

### PR DESCRIPTION
**Pull request by ableuler**
Closes #313 

Makefile / docker-compose setup of renga is ready for the proxy, don't merge until work on the proxy is finished.

Merge after: https://github.com/SwissDataScienceCenter/renku-gateway/pull/20
Merge together with https://github.com/SwissDataScienceCenter/renku-ui/pull/228

### CODE REVIEW

**Comment by ableuler** [comment]
I guess we have to do both - updating the renga charts (part of this PR) and add charts for the gateway (separate issue/PR). Or am I misunderstanding something?

**Comment by rokroskar** [comment]
yes exactly -- but my guess is that the proxy charts are not there yet? that's what I meant by opening an issue there.

**Comment by rokroskar** [comment]
yes exactly -- but my guess is that the proxy charts are not there yet? that's what I meant by opening an issue there.

**Comment by ableuler** [comment]
Your guess is 200% correct ;-)

**Comment by ableuler** [comment]
@rokroskar Helm dependencies are there now!

**Comment by ableuler** [comment]
Note that the PR with the corresponding PRs in the other repositories is currently deployed on staging.

File: charts/renku/Chart.yaml [hunk 1]
```diff
@@ -2,4 +2,4 @@ apiVersion: v1
 appVersion: '1.0'
 description: A Helm chart for the Renku platform
 name: renku
-version: 0.1.0
+version: latest
```
Comment (rokroskar) [inline charts/renku/Chart.yaml]: is this intended?

File: charts/renku/requirements.yaml [hunk 2]
```diff
@@ -18,3 +18,7 @@ dependencies:
 - name: postgresql
   version: 0.14.4
   repository: "@stable"
+- name: renku-gateway
+  alias: gateway
+  version: "0.1.0-97879d4"
+  repository: "https://swissdatasciencecenter.github.io/helm-charts/"
```
Comment (rokroskar) [inline charts/renku/requirements.yaml]: please include the gateway chart in the html template so they appear on https://swissdatasciencecenter.github.io/helm-charts/

File: docker-compose.yml [hunk 3]
```diff
@@ -294,6 +296,18 @@ services:
       review:
         aliases:
           - jupyterhub.${PLATFORM_DOMAIN}
+  proxy:
```
Comment (rokroskar) [inline docker-compose.yml]: gateway?

File: charts/renku/requirements.yaml (hunk 2 above)
Comment (ableuler) [reply charts/renku/requirements.yaml]: Done!

File: charts/renku/Chart.yaml (hunk 1 above)
Comment (ableuler) [reply charts/renku/Chart.yaml]: No - corrected.

File: docker-compose.yml (hunk 3 above)
Comment (ableuler) [reply docker-compose.yml]: Thanks for spotting this!

**Comment by rokroskar** [comment]
This looks good to me -- I assume it worked on staging. At some point soon it would be good to have a swagger spec and sequence diagrams in the documentation.

**Pull request merged**